radix.py

```diff
@@ -0,0 +1,108 @@
+def radixsort(input_list):
+    """Sort list of integers and strings with radix sort algorithm."""
+    if type(input_list[0]) is str:
+        # for case when sorting string
+        base = 256
+        string = True
+        place = 0
+    else:
+        string = False
+        base = 10  # The number of buckets to use.
+        place = 1  # IE, the 1's, 10's, or 100's place.
+    hold = -1  # A placeholder
+
+    complete = False
+    while not complete:
+        complete = True
+
+        # Declare the number of buckets to use and create them.
+        buckets = [list() for _ in xrange(base)]
+
+        # Segment the input list into appropriate buckets.
+        for item in input_list:
+            if string:
+                if place < len(item):
+                    hold = ord(item[place])
+                else:
+                    hold = 0
+            else:
+                hold = item // place
```

Comment:
I would be really great if there were a way to avoid this if/else (and the others that are to come).  What strategy might you use to solve the problem of needing to constantly branch logic to handle strings *or* numbers